backend/src/main/java/meetup/connect/meetupevent/GoogleCalendar.java

```diff
@@ -0,0 +1,204 @@
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.client.util.store.FileDataStoreFactory;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.CalendarScopes;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.EventAttendee;
+import com.google.api.services.calendar.model.EventDateTime;
+import meetup.connect.meetupevent.MeetUpEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import meetup.connect.user.User;
+import org.springframework.util.CollectionUtils;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class GoogleCalendar {
+    private static final String APPLICATION_NAME = "MeetUpConnect Calendar";
+
+    /** Global instance of the JSON factory. */
+    private static final JsonFactory JSON_FACTORY = GsonFactory.getDefaultInstance();
+
+    /** Directory to store authorization tokens for this application. */
+    private static final String TOKENS_DIRECTORY_PATH = "token";
+
+    private static final String CALENDAR_ID = "primary";
+    private static final Logger log = LoggerFactory.getLogger(GoogleCalendar.class);
+
+    /**
+     * Global instance of the scopes required by this quickstart. If modifying these scopes, delete
+     * your previously saved tokens/ folder.
+     */
+    private static final List<String> SCOPES = List.of(CalendarScopes.CALENDAR_EVENTS);
+
+    private static final String CREDENTIALS_FILE_PATH = "/credentials.json";
+    private static Calendar service;
+
+    /**
+     * Creates an authorized Credential object.
+     *
+     * @param HTTP_TRANSPORT The network HTTP Transport.
+     * @return An authorized Credential object.
+     * @throws IOException If the credentials.json file cannot be found.
+     */
+    private static Credential getCredentials(final NetHttpTransport HTTP_TRANSPORT)
+            throws IOException {
+        // Load client secrets.
+        InputStream in = GoogleCalendar.class.getResourceAsStream(CREDENTIALS_FILE_PATH);
+        if (in == null) {
+            throw new FileNotFoundException("Resource not found: " + CREDENTIALS_FILE_PATH);
+        }
+        GoogleClientSecrets clientSecrets =
+                GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+
+        // Build flow and trigger user authorization request.
+        GoogleAuthorizationCodeFlow flow =
+                new GoogleAuthorizationCodeFlow.Builder(HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES)
+                        .setDataStoreFactory(new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
```

Comment:
note: most likely you'll need `GoogleAuthorizationCodeFlow::setCredentialDataStore`